xmtp/src/builder.rs

```diff
@@ -48,28 +58,35 @@ where
                 let data_string = std::str::from_utf8(&data).map_err(|e| format!("{}", e))?;
                 let account: VmacAccount =
                     serde_json::from_str(data_string).map_err(|e| format!("{}", e))?;
-                self.account = Some(account)
+                Ok(account)
             }
             Ok(None) => {
                 let account = VmacAccount::generate();
                 let data = serde_json::to_string(&account).map_err(|e| format!("{}", e))?;
 
-                persistence.write(key, data.as_bytes())?;
+                self.persistence
+                    .as_mut()
+                    .unwrap()
+                    .write(key, data.as_bytes())?;
```

Comment:
Does the client require that that the Account is in the persistence  object? What happens if the Client is constructed directly outside of the builder?

Reply:
I would think not. The reason we write to persistence is for future client instantiations, but the client shouldn't care since it already has the account.

Reply:
Awesome.